sdk/storage/storage-file-datalake/src/StorageClient.ts

```diff
@@ -29,6 +29,7 @@ function getCoreClientOptions(pipeline: Pipeline): ExtendedServiceClientOptions
   }
   return {
     ...restOptions,
+    allowInsecureConnection: true,
```

Comment:
should we expose an option to allow customers to disable this if they want to guard against insecure connections? I suppose they could always ensure they validate the endpoint address themselves, though that might be tricky for connection strings.

Reply:
Given that by default storage accounts only allow HTTPS, we already have some protection there.  As far as I know, we haven't have customer feedback wanting this for JS or other languages. So I am inclined to not exposing an option for now.  @EmmaZhu any thoughts?

Reply:
I'd agree with Jeremy.  We can add this option when customer asks for it.